### PR TITLE
ci: generated files check

### PR DESCRIPTION
**Pull request by fadeev**
* Renamed the workflow to "Generated Files are Updated"
* Added `make specs` to the "Generated Files" check
* Moved OpenAPI generation into a separate Makefile target: `make openapi`
* Added a `make generated` target to generate all the things

### CODE REVIEW

**Comment by fadeev** [comment]
Ironically, the "Generated Files are Updated" check is failing for this PR, because https://github.com/zeta-chain/node/pull/871 has to be merged first.

File: .github/workflows/generate-files.yml [hunk 1]
```diff
@@ -17,14 +17,20 @@ jobs:
         run: |
           echo "$HOME/go/bin" >> $GITHUB_PATH
 
-      - name: Generate Go and OpenAPI
+      - name: Generate Go
```
Comment (lumtis) [inline .github/workflows/generate-files.yml:20]:
```suggestion
      - name: Generate Go code from proto files
```

File: Makefile [hunk 2]
```diff
@@ -135,15 +135,21 @@ proto:
...
-
-.PHONY: proto
+.PHONY: openapi
 
 specs:
+	@echo "--> Generating module documentation"
 	@go run ./scripts/gen-spec.go
 .PHONY: specs
 
+generated: proto openapi specs
```
Comment (lumtis) [inline Makefile:150]: Why `generated` and not `generate`?

Comment (fadeev) [reply Makefile]: Typo 🙂 5463c82e

File: .github/workflows/generate-files.yml [hunk 3]
```diff
@@ -17,14 +17,20 @@ jobs:
...
+      - name: Generate Module Documentation
+        run: make specs
+
       - name: Check for changes
         run: |
           if git diff --exit-code --ignore-space-change --ignore-all-space --ignore-cr-at-eol; then
             echo "Generated Go files are up-to-date."
           else
-            echo "::error::Generated Go and/or OpenAPI files are not up-to-date. Please run 'make proto' locally and commit any changes."
+            echo "::error::Generated files are not up-to-date. Please run 'make generated' locally and commit any changes."
```
Comment (lumtis) [inline .github/workflows/generate-files.yml:34]:
```suggestion
            echo "::error::Generated files are not up-to-date. Please run 'make generate' locally and commit any changes."
```

**Review by lumtis — APPROVED** [review]
LGTM outside of the action error. Not sure if it is related to this PR since CI currently fails for other PR as well?

**Review by brewmaster012 — APPROVED** [review]

**Comment by fadeev** [comment]
Can't merge I believe because the build task gets cancelled.

**Pull request merged**